server/server/Controllers/signingController.js

```diff
@@ -0,0 +1,101 @@
+const User = require("../Models/user");
+const bcrypt = require("bcrypt");
+const createTokens = require("../utils/createTokens");
+const { validationResult } = require("express-validator");
+
+const signingController = {
+  // controller for /v1/signIn route
+  signIn: async (req, res) => {
+    // check if user already signed in
+    if (req.user)
+      return res.status(403).send({ message: "User already logged in" });
+    const { username, password } = req.body;
+
+    // check if either username or password is missing from the request body
+    if (!username || !password)
+      return res
+        .status(400)
+        .json({ message: `Both username and password are required` });
+
+    // query database to find user with provided credentials
+    try {
+      const user = await User.findOne({ username: username });
+      if (!user)
+        return res
+          .status(400)
+          .json({ message: "User not found, recheck the provided username" });
+
+      // compare provided hash with stored hash in database
+      const passMatchResult = await bcrypt.compare(
+        password,
+        user.password_hash,
+      );
+      if (!passMatchResult)
+        return res.status(401).json({ message: "Invalid Password" });
+
+      // if all processes are passed, send back a both tokens which can be used for authentication
+      const { accessToken, refreshToken } = createTokens(user);
+
+      res
+        .cookie("accessToken", accessToken, {
+          httpOnly: true,
+          secure: false,
+          maxAge: 15 * 60 * 1000, // 15 minutes
+        })
+        .cookie("refreshToken", refreshToken, {
+          httpOnly: true,
+          secure: false,
+          maxAge: 1000 * 60 * 60 * 24 * 15, // 15 days
+        })
```

Comment:
this will work when we install the `cookie-parser` dependency in the next pull requests